src/bin/lpython.cpp

```diff
@@ -449,6 +449,151 @@ int get_errors (const std::string &infile,
     return 0;
 }
 
+bool input_in_scope_identifier(int input_location,
+                               std::pair<int, int> token_location) {
+  return token_location.first <= input_location &&
+         input_location <= token_location.second;
+}
+
+bool find_tok_in_scopes(const std::string &infile,
+                        const std::string &runtime_library_dir,
+                        CompilerOptions &compiler_options,
+                        const std::string tok_name) {
+  Allocator al(4 * 1024);
+  LFortran::diag::Diagnostics diagnostics;
+  LFortran::LocationManager lm;
+  lm.in_filename = infile;
+  std::string input = LFortran::read_file(infile);
+  lm.init_simple(input);
+  LFortran::Result<LFortran::LPython::AST::ast_t *> r1 =
+      LFortran::parse_python_file(al, runtime_library_dir, infile, diagnostics,
+                                  compiler_options.new_parser);
+  if (r1.ok) {
+    LFortran::LPython::AST::ast_t *ast = r1.result;
+    LFortran::Result<LFortran::ASR::TranslationUnit_t *> x =
+        LFortran::LPython::python_ast_to_asr(
+            al, *ast, diagnostics, true, compiler_options.disable_main,
+            compiler_options.symtab_only, infile);
+    if (!x.ok) {
+      std::cout << "{}\n";
+      return false;
+    }
+    std::vector<LFortran::LPython::document_symbols> symbol_lists;
+    LFortran::LPython::document_symbols loc;
+    for (auto &a : x.result->m_global_scope->get_scope()) {
```

Comment:
This will only be able to lookup symbols at global scope. Rather, we need to use `resolve_symbol` to lookup the symbol in the scoped symbol table. For that, we need to know which ASR node we are in.

To do that, we need to implement "index" -> ASR node calculation.

This will have to be one by the "overlapping intervals" problem, as we discussed some time ago.